packages/ra-ui-materialui/src/input/SelectInput.js

```diff
@@ -155,10 +155,21 @@ export class SelectInput extends Component {
         this.setState({ value });
     };
 
+    renderEmptyItemOption = emptyText => {
+        const { translate, translateChoice } = this.props;
+        if (React.isValidElement(emptyText))
+            return React.cloneElement(emptyText);
+        return translateChoice
```

Comment:
I think it not necessary to check for the `translateChoice` prop here as we pass the `emptyText` as the default translation anyway

Reply:
My thought was to mimic the behaviour of the other menuitem entries, let me know what behaviour you'd rather have

Reply:
Just return `translate(emptyText, { _: emptyText })` if not an element ?